### PR TITLE
Allow the "*,-*" ("no-index") pattern for destructive actions when destructive_requires_name is true

### PR DESCRIPTION
**Pull request by williamrandolph**
Since the `*,-*` index name pattern resolves to "no indices", it makes a normally destructive action into a non-destructive one. Rather than throwing a wildcards-not-allowed exception, we can allow this pattern to pass without triggering an exception. This allows the security layer to safely use a `*,-*` pattern to indicate a "no indices" result for its index resolution step, which is important because otherwise we get wildcards-not-allowed exceptions when trying to delete nonexistent concrete indices, e.g.:

```
curl -s -XDELETE 'http://elastic:changeme@localhost:9200/functional-test-actions-index?ignore_unavailable=true&pretty'
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "Wildcard expressions or all indices are not allowed"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "Wildcard expressions or all indices are not allowed"
  },
  "status" : 400
}
```

This is a fix for #67958.

I don't have the domain knowledge to know if there is a better way to avoid this issue by changing the security layer, but, if so, it would be fine by me to decline this PR in favor of that approach.

### CODE REVIEW

**Comment by williamrandolph** [comment]
Todo:
- [ ] fix test failures
- [ ] order matters with wildcards, so check for `*,-*` exactly
- [ ] make sure all destructive actions have test coverage for this issue

**Comment by williamrandolph** [comment]
I created https://github.com/elastic/elasticsearch/issues/68079 for the test failure.

**Comment by williamrandolph** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by williamrandolph** [comment]
@elasticmachine run elasticsearch-ci/2

File: server/src/main/java/org/elasticsearch/action/support/DestructiveOperations.java [hunk 1]
```diff
@@ -59,6 +61,9 @@ public void failDestructive(String[] aliasesOrIndices) {
             if (hasWildcardUsage(aliasesOrIndices[0])) {
                 throw new IllegalArgumentException("Wildcard expressions or all indices are not allowed");
             }
+        } else if (Arrays.equals(aliasesOrIndices, new String[]{"*", "-*"})) {
```
Comment (jaymode) [inline server/src/main/java/org/elasticsearch/action/support/DestructiveOperations.java:64]: Can you make the String[] a static final variable?

Comment (jaymode) [reply server/src/main/java/org/elasticsearch/action/support/DestructiveOperations.java:64]: Also you could simplify the code and remove the else below by making the check Arrays.equals() == false. Either way is fine

**Review by jaymode — APPROVED** [review]
LGTM

**Comment by williamrandolph** [comment]
Backport commit: ddda5f2e930270e4607315e6c2bec5b7f89242b5

**Pull request merged**